src/services/authentication/strategies/TSPStrategy.js

```diff
@@ -0,0 +1,115 @@
+const { AuthenticationBaseStrategy } = require('@feathersjs/authentication');
+const { NotAuthenticated } = require('@feathersjs/errors');
+const { omit } = require('lodash');
+//const { jose } = require('jose');
+
+class TSPStrategy extends AuthenticationBaseStrategy {
+
+	get configuration() {
+
+		const authConfig = this.authentication.configuration;
+		const config = super.configuration || {};
+		return {
+			service: authConfig.service,
+			entity: authConfig.entity,
+			entityId: authConfig.entityId,
+			errorMessage: 'Invalid login',
+			entityUsernameField: config.usernameField,
+			...config,
+		};
+	}
+
+	verifyConfiguration() {
+
+		const config = this.configuration;
+		['usernameField'].forEach((prop) => {
+			if (typeof config[prop] !== 'string') {
+				throw new Error(`'${this.name}' authentication strategy requires a '${prop}' setting`);
+			}
+		});
+	}
+
+	async authenticate(authentication, params) {
+
+		const { app } = this;
+		const user = await app.service('users').get(authentication.ticket);
+		const { entity } = this.configuration;
+		const result = await this.findEntity(authentication.ticket, omit(params, 'provider'));
+		return {
+			authentication: { strategy: this.name },
+			[entity]: await this.getEntity(result, params),
+		};
+	}
+
+	/**
+	 *
+	 * @param ticket
+	 */
+	decryptTSPTicket(ticket) {
+
+		// Get Payload out of the Ticket
+		const payload = new Buffer(ticket.split('.')[1], 'base64').toString();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [new Buffer() is deprecated. Use Buffer.from(), Buffer.alloc(), or Buffer.allocUnsafe() instead. (no-buffer-constructor)](https://app.codacy.com/app/schul-cloud/schulcloud-server/pullRequest?prid=4491705)